ibis/backends/base/sql/alchemy/__init__.py

```diff
@@ -43,30 +54,369 @@
 )
 
 
+class _AutoCloseCursor:
+    """
+    Wraps a SQLAlchemy ResultProxy and ensures that .close() is called on
+    garbage collection
+    """
+
+    def __init__(self, original_cursor):
+        self.original_cursor = original_cursor
+
+    def __del__(self):
+        self.original_cursor.close()
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, type, value, tb):
+        self.original_cursor.close()
+
+    def fetchall(self):
+        return self.original_cursor.fetchall()
+
+
 class BaseAlchemyBackend(BaseSQLBackend):
     """
     Base backend class for backends that compile to SQL with SQLAlchemy.
     """
 
     database_class = AlchemyDatabase
     table_class = AlchemyTable
+    compiler = AlchemyCompiler
+    has_attachment = False
 
     @property
     def version(self):
-        return '.'.join(map(str, self.client.con.dialect.server_version_info))
+        return '.'.join(map(str, self.con.dialect.server_version_info))
 
     def list_tables(self, like=None, database=None):
-        inspector = sqlalchemy.inspect(self.client.con)
+        inspector = sqlalchemy.inspect(self.con)
         tables = inspector.get_table_names(
             schema=database
         ) + inspector.get_view_names(schema=database)
         return self._filter_with_like(tables, like)
 
-    @property
-    def current_database(self):
-        return self.client.database_name
-
     def list_databases(self, like=None):
         """List databases in the current server."""
-        databases = self.client.inspector.get_schema_names()
+        databases = self.inspector.get_schema_names()
         return self._filter_with_like(databases, like)
+
+    @property
+    def inspector(self):
+        self._inspector.info_cache.clear()
+        return self._inspector
+
+    @staticmethod
```

Comment:
we typically don't use staticmethods as this is an anti-pattern.

Reply:
I think this a carryover from the previous location. `staticmethod`s are useful in cases where an otherwise free function is specific to an operation in a class, but doesn't need the instance (`self`) or the type (`cls`) of the class to do its job. I personally think it's fine to leave as a `staticmethod`.